core/server/src/test/java/alluxio/master/AlluxioMasterRestServiceHandlerTest.java

```diff
@@ -258,4 +277,168 @@ public void getWorkerInfoList() {
     assertFalse(entry.isEmpty());
     assertEquals(mockWorkerInfo.getId(), entry.get(0).getId());
   }
+
+  public static class UnderFileSystemFactoryMock implements UnderFileSystemFactory {
+
+    @Override
+    public UnderFileSystem create(String path, Object ufsConf) {
+      return new UnderFileSystemMock(new AlluxioURI(path));
+    }
+
+    @Override
+    public boolean supportsPath(String path) {
+      return path.startsWith("test");
+    }
+  }
+
+  public static class UnderFileSystemMock extends UnderFileSystem {
```

Comment:
Instead of creating this class, could we create a mock of `UnderFileSystem` which only mocks the `getSpace` method? Then we could pass it to the constructor of `UnderFileSystemFactoryMock`, e.g.

```java
public UnderFileSystemFactoryMock(UnderFileSystem ufs, String pathPrefix) {
  mUfs = ufs;
  mPathPrefix = pathPrefix
}

@Override
public UnderFileSystem create(String path, Object ufsConf) {
  return mUfs;
}

@Override
public boolean supportsPath(String path) {
  return path.startsWith(mPathPrefix);
}
```